lib/src/shim.dart

```diff
@@ -302,9 +303,34 @@ class AnalysisServerAdapter extends LanguageServer {
       if (hovers.isEmpty) return null;
       var hover = hovers.first;
       var range = rangeFromOffset(_files[path], hover.offset, hover.length);
-      return new Hover((b) => b
-        ..contents = _hoverMessage(hover)
-        ..range = range);
+
+      final List<String> clientFormats =
+          clientCapabilities?.textDocument?.hover?.contentFormat;
+
+      final List<String> sharedFormats =
```

Comment:
this won't work in Dart 2. Needs to be either `Iterable<String>` or include a `.toList()` call. Better yet omit the type on the left hand side altogether.

Reply:
Doh (again!)